apps/remote_control/lib/lexical/remote_control/project_node.ex

```diff
@@ -0,0 +1,83 @@
+defmodule Lexical.RemoteControl.ProjectNode do
+  alias Lexical.RemoteControl
+  require Logger
+
+  defmodule State do
+    defstruct [:project, :paths, :cookie]
+
+    def new(project) do
+      paths = format_prepending_paths(RemoteControl.glob_paths())
+      cookie = Node.get_cookie()
+
+      struct!(__MODULE__, paths: paths, cookie: cookie, project: project)
+    end
+
+    defp format_prepending_paths(paths_as_charlists) do
+      Enum.map_join(paths_as_charlists, " -pa ", &Path.expand/1)
+    end
+  end
+
+  alias Lexical.RemoteControl.ProjectNodeSupervisor
+
+  use GenServer
+
+  def wait_until_started(project, project_listener, boot_timeout \\ 5_000) do
+    :ok = :net_kernel.monitor_nodes(true, node_type: :visible)
```

Comment:
the genserver should be monitoring nodes when it inits, otherwise messages will be delivered to whatever process calls this function.